### PR TITLE
[DatePicker] Change OnSelectedDateAsync logic

### PR DESCRIPTION
**Pull request by MarvinKlein1508**
# Pull Request

## 📖 Description

This fixes the issue that the `FluentDatePicker` does not work, when the value of `@bind` is null. The issue comes from the fact that  
`(value ?? DateTime.MinValue).Date + Value?.TimeOfDay` returns always null when `Value is null`

### 🎫 Issues

* #2197

### General

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fluentui-blazor/blob/master/docs/contributing.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: src/Core/Components/DateTime/FluentDatePicker.razor.cs [hunk 1]
```diff
@@ -66,10 +66,15 @@ protected async Task OnSelectedDateAsync(DateTime? value)
     {
         Opened = false;
 
-        var updatedValue = Value?.TimeOfDay != TimeSpan.Zero
+        DateTime? updatedValue = value;
+
+        if (Value is not null && value is not null)
+        {
+            updatedValue = Value?.TimeOfDay != TimeSpan.Zero
             ? (value ?? DateTime.MinValue).Date + Value?.TimeOfDay
```
Comment (dvoituron) [inline src/Core/Components/DateTime/FluentDatePicker.razor.cs:74]: Why do you check `value` with `??` if the previous test already that?

**Review by dvoituron — CHANGES_REQUESTED** [review]
Could you add a UnitTest to validate this particular case?

File: src/Core/Components/DateTime/FluentDatePicker.razor.cs (hunk 1 above)
Comment (MarvinKlein1508) [reply src/Core/Components/DateTime/FluentDatePicker.razor.cs]: my bad. I removed it :)

**Comment by MarvinKlein1508** [comment]
@dvoituron added UNIT-Test

**Review by dvoituron — APPROVED** [review]

**Pull request merged**